dask/array/slicing.py

```diff
@@ -364,11 +364,15 @@ def _slice_1d(dim_shape, lengths, index):
 
     And negative slicing
 
-    >>> _slice_1d(100, [20, 20, 20, 20, 20], slice(100, 0, -3))
-    {0: slice(-2, -20, -3), 1: slice(-1, -21, -3), 2: slice(-3, -21, -3), 3: slice(-2, -21, -3), 4: slice(-1, -21, -3)}
+    >>> exp_res = {0: slice(-2, -20, -3), 1: slice(-1, -21, -3), 2: slice(-3, -21, -3),
+    ...            3: slice(-2, -21, -3), 4: slice(-1, -21, -3)}
+    >>> _slice_1d(100, [20, 20, 20, 20, 20], slice(100, 0, -3)) == exp_res
+    True
 
-    >>> _slice_1d(100, [20, 20, 20, 20, 20], slice(100, 12, -3))
-    {0: slice(-2, -8, -3), 1: slice(-1, -21, -3), 2: slice(-3, -21, -3), 3: slice(-2, -21, -3), 4: slice(-1, -21, -3)}
+    >>> exp_res = {0: slice(-2, -8, -3), 1: slice(-1, -21, -3), 2: slice(-3, -21, -3),
+    ...            3: slice(-2, -21, -3), 4: slice(-1, -21, -3)}
+    >>> _slice_1d(100, [20, 20, 20, 20, 20], slice(100, 12, -3)) == exp_res
+    True
```

Comment:
Would it be possible to include full results here in the output rather than explicitly test?  In Python 3.6 we can probably depend on deterministic dict ordering.  We could probably also use the `# doctest: +NORMALIZE_WHITESPACE` pragma to help improve the styling, perhaps something like the following?

```python

>>> _slice_1d(100, [20, 20, 20, 20, 20], slice(100, 12, -3))  # doctest: +NORMALIZE_WHITESPACE
{0: slice(-2, -8, -3), 
 1: slice(-1, -21, -3), 
 2: slice(-3, -21, -3),
 3: slice(-2, -21, -3), 
 4: slice(-1, -21, -3)}
￼

Reply:
I originally used the `# doctest: +NORMALIZE_WHITESPACE` pragma, but in the end I stuck to the doctests documentation workaround: https://docs.python.org/3.6/library/doctest.html#warnings

I agree that with full results and pragma we can improve readability and styling, I will have a look and I will modify this part.